marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/getOpticPlan.sjs

```diff
@@ -58,6 +80,15 @@ structuredQuery = fn.head(xdmp.unquote(structuredQuery)).root;
 searchText = searchText || '';
 queryOptions = fn.head(xdmp.unquote(queryOptions)).root;
 
+/*
+ * Filtering out columns (properties) that are of object/array type since we don't support them for now.
+ * Also replacing hyphen with underscore for column names (entity property names), schema names and view names since TDE's do the same.
+ */
+const simplePropertySet = filterObjectAndArrayTypeProperties(schemaName);
+columns = columns.filter(column => simplePropertySet.has(column)).map(column => replaceHyphenWithUnderscore(column));
+viewName = replaceHyphenWithUnderscore(viewName);
+schemaName = replaceHyphenWithUnderscore(schemaName);
+
 const newOptions = fn.head(xdmp.xsltEval(stylesheet, queryOptions)).root;
```

Comment:
This is a tangent to this PR, but what is the minimum role assigned to the user that should be able to use this data service? I believe we'll need a ticket to add the xslt-eval privilege to that role as part of another ticket.

Reply:
Good point. 
This means we need to add the xslt-eval privilege to hub-central-entity-exporter role.

Reply:
Added a ticket: https://project.marklogic.com/jira/browse/DHFPROD-5139